cfme/tests/automate/test_file_import.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from cfme import test_requirements
+
+pytestmark = [test_requirements.automate, pytest.mark.tier(3)]
+
+
+def test_domain_import_file(import_datastore):
+    """This test case Verifies that a domain can be imported from file.
+
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/6h
+        caseimportance: medium
+        startsin: 5.7
+        casecomponent: Automate
+        tags: automate
+        testSteps:
+            1. Navigate to Automation > Automate > Import/Export
+            2. Upload zip datastore file
+            3. Select domain which like to import
+        expectedResults:
+            1.
+            2.
+            3. Import should work. Check imported or not.
+    """
+    domain = import_datastore(file_name="bz_1715396.zip", from_domain="BZ_1715396")
```

Comment:
I'm not sure this is creating a great testing pattern here, returning a function to be called by the test itself.

Consider a bit different design here, where `import_datastore` fixture actually behaves as a fixture, and depends on parametrized `file_name` and `from_domain`.  

Tests using this global fixture can then parametrize at the appropriate scope. They can skip in setup when the expected file isn't available on the server.  They can include file/domain metadata in the id for the parameter, resulting in test functions that are specific to the files used to run the test.

Reply:
Yup, I agreed but as per my finding not good in this case.

> I'm not sure this is creating a great testing pattern here, returning a function to be called by the test itself.

It is one of standard way to implement `fixture` taking `args` 
https://docs.pytest.org/en/latest/fixture.html#factories-as-fixtures

> Consider a bit different design here, where `import_datastore` fixture actually behaves as a fixture, and depends on parametrized `file_name` and `from_domain`.

:+1:  @mshriver I tried this implementation; its good way when we are checking multiple `datastore` files. This is not our requirement. even if we try to implement it we need fake parametrization for single item :D (eg. current test using one file). 

> Tests using this global fixture can then parametrize at the appropriate scope. They can skip in setup when the expected file isn't available on the server. They can include file/domain metadata in the id for the parameter, resulting in test functions that are specific to the files used to run the test.
 
So, other way of implementation is:
```
@pytest.fixture()
def import_datastore(appliance, file_name, from_domain, to_domain):
    pass

@pytest.mark.parametrize("file_name", ["niz.zip"])
@pytest.mark.parametrize("from_domain", ["abc_domain"])
@pytest.mark.parametrize("to_domain", ["xyz_domain"])
def test_import(appliance, file_name, from_domain, to_domain, import_datastore):
    pass
```
- both method will have scope limitation
- second need fake parametrization (which going to be use mostly)
- In first method also parametrization possible (when needed)

If you ack... I will go with second way as you suggested. :+1:

Reply:
I agree with above reviewers. above fixture and its usage look a bit weird. please give above suggessions some thought